addressfield.components.inc

```diff
@@ -0,0 +1,486 @@
+<?php
+/**
+ * @file
+ * Webform Component information for an address field type.
+ * 
+ * Adapted from Drupal 7's module addressfield_tokens
+ * 
+ */
+
+/**
+ * Specify the default properties of a component.
+ *
+ * @return array
+ *   An array defining the default structure of a component.
+ */
+function _webform_defaults_addressfield() {
+  return array(
+    'name'      => '',
+    'form_key'  => NULL,
+    'required' => 0,
+    'pid'       => 0,
+    'weight'    => 0,
+    'extra'     => array(
+      'title_display'             => 0,
+      'private'                   => FALSE,
+      'attributes'                => array(),
+      'description'               => '',
+      'available_countries'       => array(),
+      'default_country'           => '',
+      'format_handlers'           => array(),
+      'csv_separate'              => 0,
+    ),
+  );
+}
+
+/**
+ * Generate the form for editing a component.
+ *
+ * Create a set of form elements to be displayed on the form for editing this
+ * component. Use care naming the form items, as this correlates directly to the
+ * database schema. The component "Name" and "Description" fields are added to
+ * every component type and are not necessary to specify here (although they
+ * may be overridden if desired).
+ *
+ * @param mixed $component
+ *   A Webform component array.
+ *
+ * @return array
+ *   An array of form items to be displayed on the edit component page
+ */
+function _webform_edit_addressfield($component) {
+  $form = array();
+
+  $form['extra']['available_countries'] = array(
+    '#type' => 'select',
+    '#multiple' => TRUE,
+    '#title' => t('Available countries'),
+    '#description' => t('If no countries are selected, all countries will be available.'),
+    '#options' => _addressfield_country_options_list(),
+    '#default_value' => $component['extra']['available_countries'],
+  );
+  $form['extra']['default_country'] = array(
+    '#type' => 'select',
+    '#multiple' => FALSE,
+    '#title' => t('Default country'),
+    '#description' => t('Select which country should be selected as the default.'),
```

Comment:
`Default country` is self-explanatory, I would remove this description I don't think it adds much value.

Reply:
Done!